src/audioplayer.rs

```diff
@@ -63,13 +65,20 @@ impl AudioPlayer{
     }
 
     pub fn set_station(&mut self, station: Station){
-        let station_url = self.client.get_playable_station_url(&station);
         Self::update(&self.update_callbacks, Update::Station(station.clone()));
         Self::update(&self.update_callbacks, Update::Title("".to_string()));
-        self.station = Some(station);
-
         self.playbin.set_state(gstreamer::State::Null);
-        self.playbin.set_property("uri", &station_url);
+        self.station = Some(station);
+        
+        //request url and set it in a new thread
+        let playbin = Mutex::new(self.playbin.clone());
+        let client = self.client.clone();
+        let station = self.station.unwrap().clone(); 
```

Comment:
I've renamed them, becuase as far I know this aren't real atomic types.[ More information here](https://doc.rust-lang.org/std/sync/atomic/index.html)